packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-processing-job.ts

```diff
@@ -0,0 +1,546 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Duration, Lazy, Size, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+import {
+  AppSpecification,
+  DatasetDefinition,
+  ExperimentConfig,
+  InputMode,
+  NetworkConfig,
+  ProcessingInput,
+  ProcessingOutputConfig,
+  ProcessingResources,
+  S3DataType,
+  CompressionType,
+  StoppingCondition,
+  S3DataDistributionType,
+  AthenaOutputFormat,
+  RedshiftOutputFormat,
+  S3UploadMode,
+} from './base-types';
+
+/**
+ * Properties for creating an Amazon SageMaker processing job task
+ */
+export interface SageMakerCreateProcessingJobProps extends sfn.TaskStateBaseProps {
+  /**
+   * Role for the Processing Job. The role must be granted all necessary permissions for the SageMaker processing job to
+   * be able to operate.
+   *
+   * See https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-createprocessingjob-perms
+   *
+   * @default - a role with appropriate permissions will be created.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Configures the processing job to run a specified Docker container image.
+   */
+  readonly appSpecification: AppSpecification;
+
+  /**
+   * The environment variables (map of keys to values) to set in the Docker container. Up to 100 key and values entries
+   * in the map are supported. Keys and values must be strings with a maximium length of 256.
+   *
+   * @default - No environment variables
+   */
+  readonly environment?: sfn.TaskInput;
```

Comment:
This decision might be controversial (making the environment variables Task input), but I'm interested to see your thoughts in this specific case and to see if you might have the same thought process when considering other properties.

Today, in similar constructs, the equivalent property is not a task input (and instead is a string to string map). An example elsewhere could be found in the [transformation job implementation](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts#L34-L39) or in the [training job implementation](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts#L92-L94).

However, I think we can potentially make a case for allowing the environment variables to be controlled dynamically in the processing job. In the future, we can potentially migrate the other sagemaker tasks' environment variables to also be task input as well.

Here's my thought process: These environment variables are a great mechanism for controlling behavior within a docker container. Although this example is not perfectly analogous, this might illustrate where I'm trying to go with this:

In a previous (now deprecated) implementation of invoking lambdas as step functions tasks, the payload to lambdas was completely static (similar to how the environment variables are in the current sagemaker tasks): [static lambda payload invocation](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke-function.ts#L8-L18). However, much flexibility can be enjoyed if the payload can be dynamic using task input, further controlling the behavior of your lambda function. Thus, in the current revision of lamda invocation tasks, the [lambda payload invocation is now dynamic](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts#L18-L23) leveraging task inputs.

I think the same case definitely applies here. While lambda payload is definitely more flexible than just the environment variables in this case, I think there are many cases where it's useful to have environment variables be different between executions of a particular step functions state machine and allowing application logic to dictate it as an input. Additionally, if a user just wants the environment variables to just be static, they can still do so with task input.

I'm unsure of how much further this idea can or should be applied (in terms of other properties of this construct). I've applied it to tags as well, but please do let me know if you feel different.